src/Directory.Build.targets

```diff
@@ -47,7 +47,7 @@
       <_RunOnCore Condition="$(TargetFramework.StartsWith('netcoreapp'))">true</_RunOnCore>
       
       <_UseOpenCover>false</_UseOpenCover>
-      <_UseOpenCover Condition="'$(CODECOV_TOKEN)' != '' and '$(_RunOnCore)' != 'true'">true</_UseOpenCover>
+      <_UseOpenCover Condition="'$(CODECOV_TOKEN)' != '' and '$(Configuration)' == 'Debug' and '$(_RunOnCore)' != 'true'">true</_UseOpenCover>
```

Comment:
:memo: I would prefer this change not be applied this far inside the implementation. Code coverage for release builds works fine and it will be confusing to arbitrarily block it here.

Reply:
This is what I get without this change
```
Failed to register(user:True,register:True,is64:False):5 the profiler assembly; you may want to look into permissions or using the -register:user option instead. C:\Windows\system32\regsvr32.exe /s  /n /i:user "D:\j\workspace\Windows_NT_Re---c6b3d408\.packages\opencover\4.6.519\tools\x86\OpenCover.Profiler.dll"
If you are unable to resolve the issue please contact the OpenCover development team
see https://www.github.com/opencover/opencover/issues 
```

Reply:
Ah, that's an unrelated race condition. It can occur if you try to spawn instances of OpenCover concurrently. In dotnet/roslyn, I avoided the issue by running RunTests.exe through it (one top level instance). In dotnet/roslyn-analyzers, I avoided the issue by running a PDB extraction process that resulted in pseudo-randomization of the initialization time for test runner processes. If you made a change that eliminated the PDB extraction process, this issue is likely to reappear and we'll need to address it. It's not a debug- or release-specific issue.

Reply:
I see. I have not made the change in PDB extraction. It's still converting. For some reason it usually fails in Release and not in Debug.

Reply:
Is there an issue tracking the race condition? Sounds like a major problem.

Reply:
@sharwell @tmat, what is the best way to unblock the queues here? If the correct fix is non-trivial, I am fine with merging this PR to unblock the queues, and then have a separate PR to correctly resolve it.

Reply:
Please merge this PR. It doesn't change the logic that was previously implemented. We have previously not run code coverage for release builds.